src/executor/plan_executor.py

```diff
@@ -106,7 +107,11 @@ def execute_plan(self):
             execution_tree.exec()
         else:
             for batch in execution_tree.exec():
-                output_batches += batch
+                if not batch.empty():
```

Comment:
We can move this check to the _add_ function

Reply:
Move to the add function.